### PR TITLE
CORTX-29266 Provisioner Sanity Pipeline

### PR DESCRIPTION
**Pull request by aayushisharma88**
Signed-off-by: Aayushi Sharma <aayushi.sharma@seagate.com>

# Problem Statement
There was no sanity for provisioning.

# Design
- We created a sanity pipeline.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] https://eos-jenkins.colo.seagate.com/job/Provisioner/job/Prvsnr-Sanity-Test/69/console
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/automation/kubernetes/nightly-prvsn-sanity-deploy.groovy [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+                }
+            }
+        }
+        stage('Destroy CORTX Cluster') {
+            steps {
+                script {
+                    remotes = remote_host()
+                    sshCommand remote: remotes[0], command: """
+                        cd ${WORK_SPACE}
```
Comment (gauravchaudhari02) [inline jenkins/automation/kubernetes/nightly-prvsn-sanity-deploy.groovy:88]: I guess we are in `WORKSPACE` directory everytime, so can you try running job by removing `cd ${WORKSPACE}` in each stage where it is used?

File: jenkins/automation/kubernetes/nightly-prvsn-sanity-deploy.groovy [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+            }
+        }
+
+        stage('Checkout CORTX Script') {
+            steps {
+                script {
+                    remotes = remote_host()
+                    for (remote in remotes) {
+                        sshCommand remote: remote, command: """
+                            cd /root
```
Comment (gauravchaudhari02) [inline jenkins/automation/kubernetes/nightly-prvsn-sanity-deploy.groovy:115]: can you use pushd/popd instead of cd here? as used on line 138?

**Comment by gauravchaudhari02** [comment]
@aayushisharma88, please fill template with problem statement and how you resolved it in design section. Add tested job link as well.

**Review by gauravchaudhari02 — DISMISSED** [review]

**Review by shailesh-vaidya — COMMENTED** [review]
Parameters from code and in job provided - https://eos-jenkins.colo.seagate.com/job/Provisioner/job/Prvsnr-Sanity-Test/69/console look different. Pipeline is running from `https://github.com/Seagate/cortx-test/tree/CICD-Pipeline`  
Can you please try running the job using code from the proposed changes?

**Review by shailesh-vaidya — APPROVED** [review]

**Comment by shailesh-vaidya** [comment]
@gauravchaudhari02 - Please review.  We will merge this after https://github.com/Seagate/cortx-re/pull/935

**Review by gauravchaudhari02 — DISMISSED** [review]

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request closed without merging**